test/index.test.ts

```diff
@@ -20,121 +21,127 @@ describe('gcpMetadata', () => {
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
   });
 
-  it('should return error if statusCode is not 200', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should return error if statusCode is not 200', async () => {
     nock(HOST).get(PATH).reply(418, {}, {'Metadata-Flavor': 'Google'});
-    getMetadata((err) => {
-      assert(err instanceof Error);
-      done();
-    });
+    try {
+      await gcp.project();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
   });
 
-  it('should retry if the initial request fails', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should retry if the initial request fails', async () => {
     nock(HOST).get(`${PATH}/${TYPE}`).reply(500);
     nock(HOST).get(`${PATH}/${TYPE}`).reply(200, {});
-    getMetadata((err, res) => {
-      assert.equal(res!.config.url, `${BASE_URL}/${TYPE}`);
-      assert.equal(res!.config.headers['Metadata-Flavor'], 'Google');
-      done();
-    });
+    const res = await gcp.instance();
+    assert.equal(res.config.url, `${BASE_URL}/${TYPE}`);
+    assert.equal(res.config.headers['Metadata-Flavor'], 'Google');
   });
 
-  it('should throw if request options are passed', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
-    // tslint:disable-next-line no-any
-    (gcpMetadata as any).instance({qs: {one: 'two'}}, (err: Error) => {
-      assert.notEqual(err, null);
-      assert(err.message.startsWith('\'qs\' is not a valid'));
-      done();
-    });
+  it('should throw if request options are passed', async () => {
+    try {
+      // tslint:disable-next-line no-any testing validation
+      await (gcp as any).instance({qs: {one: 'two'}});
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e.message.startsWith('\'qs\' is not a valid'));
+    }
   });
 
-  it('should not retry on DNS errors', (done) => {
-    const getMetadata = gcpMetadata._buildMetadataAccessor(TYPE);
+  it('should not retry on DNS errors', async () => {
     nock(HOST).get(PATH).replyWithError({code: 'ETIMEDOUT'});
     nock(HOST).get(PATH).reply(200, {});
-    getMetadata(err => {
-      assert(err instanceof Error);
-      done();
+    try {
+      await gcp.project();
+      assert.fail('Expected to throw');
+    } catch (e) {
+      assert(e instanceof Error);
+    }
+  });
+
+  it('should use HEAD to find the metadata server', async () => {
+    nock(HOST).head(`${PATH}/instance`).reply(200, null, {
+      'metadata-flavor': 'Google'
     });
+    const isAvailable = await gcp.isAvailable();
+    assert.equal(isAvailable, true);
+  });
+
+  it('should throw if the isAvailable check returns a 500', async () => {
+    nock(HOST).head(`${PATH}/instance`).times(4).reply(500);
+    try {
+      await gcp.isAvailable();
+      assert.fail('Expected to throw');
```

Comment:
nit: This assert will be caught and the error message reported to the developer would not immediately point to the correct assertion that failed.